consent-ws/src/main/java/org/broadinstitute/consent/http/service/DatabaseConsentAPI.java

```diff
@@ -230,30 +230,44 @@ public void logicalDelete(String id) throws UnknownIdentifierException {
         return getAllAssociationsForConsent(consentId);
     }
 
-    /** Can't add this to the DAO interface =( **/
-    private List<String> updateAssociations(String consentId, String associationType, List<String> ids){
+    /**
+     * Can't add this to the DAO interface =(
+     **/
+    private List<String> updateAssociations(String consentId, String associationType, List<String> ids) {
         Handle h = jdbi.open();
-        PreparedBatch insertBatch = h.prepareBatch("insert into consentassociations (consentId, associationType, objectId) values (?, ?, ?)");
-        for(String id: ids){
-            insertBatch.add(consentId, associationType, id);
+        PreparedBatch insertBatch = h.prepareBatch("insert into consentassociations (consentId, associationType, dataSetId) values (?, ?, ?)");
+        for (String id : ids) {
+            insertBatch.add(consentId, associationType, dataSetDAO.findDataSetIdByObjectId(id));
         }
         List<Long> insertedIds = insertBatch.executeAndGenerateKeys(LongColumnMapper.PRIMITIVE).list();
         h.close();
         List<String> stringsList = new ArrayList<>();
-        for (Long id: insertedIds) stringsList.add(id.toString());
+        for (Long id : insertedIds) stringsList.add(id.toString());
         return stringsList;
     }
 
-    private void validateElements(List<String> newIds) {
+    private void validateEmptyObjectIds(List<String> newIds) {
         if (CollectionUtils.isNotEmpty(newIds)) {
             newIds.stream().forEach(objectId -> {
-                if(StringUtils.isEmpty(objectId)){
+                if (StringUtils.isEmpty(objectId)) {
                     throw new IllegalArgumentException("Elements are required");
                 }
             });
         }
     }
 
+    private void processAssociation(List<String> objectIds) {
+        if (CollectionUtils.isNotEmpty(objectIds)) {
+            objectIds.stream().forEach(objectId -> {
+                Integer dsId = dataSetDAO.findDataSetIdByObjectId(objectId);
+                if (dsId == null) {
+                    dataSetDAO.insertAll(Arrays.asList(new DataSet(objectId)));
+                }
+
+            });
```

Comment:
two dao calls in a loop make me nervous. Could these be re-written as a query for all datasets not in <objectIds> and then bulk adding?